### PR TITLE
Backports for 1.8-rc1/beta4

### LINKED ISSUE #44732: Dispatch failure or failure to throw error regression
Code to repro the issue:

```jl
julia> using Dash

julia> app = dash(assets_folder = "assets_compressed", compress = true);

julia> make_handler(app)
```

In 1.7:

```jl
julia> make_handler(app)
ERROR: The layout must be a component, tree of components, or a function which returns a component.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] validate_layout(layout::Nothing)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:44
 [3] make_handler(app::Dash.DashApp, registry::DashBase.ResourcesRegistry; check_layout::Bool)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:123
 [4] make_handler(app::Dash.DashApp)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:158
```

In 1.8:

```jl
julia> make_handler(app); # No error
```

Now, this should cleary error because what ends up getting called is

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/src/handler/make_handler.jl#L123

which has these three definitions:

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/src/handler/make_handler.jl#L40-L44

and `app.layout` is `nothing`:

```jl
julia> app.layout === nothing
true
```

In fact, it errors if we execute it manually:

```jl
julia> Dash.validate_layout(Dash.get_layout(app))
ERROR: The layout must be a component, tree of components, or a function which returns a component.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] validate_layout(layout::Nothing)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:44
```

and also when running it through the debugger:

```jl
julia> using Debugger

julia> @run make_handler(app)
ERROR: The layout must be a component, tree of components, or a function which returns a component.
````

So something here makes the dispatch go to the wrong function or the error to be swallowed.

### LINKED ISSUE #44735: typeintersect Vararg aborts
this is not good:
```
julia> typeintersect(Vararg{Int}, Int)
julia: /data/vtjnash/julia/src/subtype.c:1261: subtype: Assertion `!jl_is_vararg(x) && !jl_is_vararg(y)' failed.

signal (6): Aborted
```
and other mild oddities (I think these are correct, since they can appear as type parameters, they should be valid as arguments also)
```
julia> typeintersect(1, 2)
Union{}

julia> typeintersect(1, 1)
1

julia> typeintersect(1, Any)
1

julia> typeintersect(Any, 1)
1
```

### LINKED ISSUE #44763: missing effects propagation from `Core.invoke` call
```julia
julia> invoke_println(x) = Base.@invoke println(x::Any)
invoke_println (generic function with 1 method)

julia> code_typed(; optimize=false) do
           invoke_println(42)
       end;
42
```

So `invoke_println` is wrongly concrete-evaled.

### LINKED ISSUE #45024: lost `expected assignment after "const"` error
```
julia> const x
ERROR: `body` expression must terminate in `return`. Use `block` instead.

julia> const x::Int
# allowed
```

These both used to give the "expected assignment" error. This is because we need to parse `const x` inside struct definitions, and `const x` is a valid IR form but not generally allowed as input syntax. We need to either fix the error, or decide to allow `const x` (which declares `x` constant without giving it a value, which is a state we didn't want to be observable for too long).

### LINKED ISSUE #45162: Using keyword argument prevents specialization
I raised this issue [on discourse](https://discourse.julialang.org/t/using-a-keyword-argument-leads-to-enormous-allocations/80354), where the consensus seems to be that this is at least very confusing, and possibly a bug.

If I *use* a kwarg for a function that was passed as an argument to another function, Julia does not specialize the latter function.  If I don't use the kwarg for that very same function, it does specialize.  Below, I'll include a less trivial example that's closer to my actual use case.  But schematically, the idea is this:
```julia
f1(a; b=10) = a+b
f2(c, f) = c + f(20)
f3(d, f) = d + f(30; b=40)
```
Calling, for example, `f2(5, f1)` will result in a specialized `f2`; calling `f3(5, f1)` will not result in specialization.  I imagine Julia is clever enough to optimize the problem away in this schematic.  But in my code, I was seeing slowdowns of ~100x and allocations of multiple GiBs on each call to my core computation function.

As pointed out in discourse, it's possible to manually trigger specialization by adding a type parameter.  But the failure to *automatically* specialize is a problem for a few reasons:

1. It's surprising.  The [performance tip on specialization](https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing) says "Julia will always specialize when the argument is used within the method, but not if the argument is just passed through to another function."  In this case, I *did* use the argument (the function with the kwarg).  From the discussion on discourse, it looks like the problem is that Julia immediately lowers that to just pass the function through to `Core.kwfunc`.  So technically the argument "is just passed through to another function" — but not by the programmer.  (Gotta love passive voice!)
2. It's very hard to diagnose.  None of the usual tools — profiling, allocation tracking, `@code_warntype`, JET, Traceur — pointed out any problem with the use of kwargs.  In fact, profiling and allocation actively focused my attention on other parts of the code that were not at all the source of the problem.  Even the `(@which f(...)).specializations` trick from that section of the performance tips seemed to say the function *was* being specialized for my arguments.  (See below.)
3. It seems to contradict the docs.  If the goal when designing this heuristic is to detect when a function is "just passed through" so that it will "usually [have] no performance impact at runtime", surely the decision of how to arrange parameters in a function definition should not affect the result.

So, at the very least, I would think this is a documentation bug, because the kwarg wrinkle should be noted in that performance tip — rather than requiring the user to mentally combine disparate arcana from the most cryptic parts of the docs.  It would also be nice if some standard tools could point toward the source of the problem.  But maybe this is truly a bug in Julia, which should actually specialize even when a kwarg is used?

---

For reference, here's a working example that's complicated enough that Julia doesn't just optimize the problem away, while still being a greatly simplified version of my actual use case:
```julia
using Profile

function index(n, mp, m; n_max=n)
    n + mp + m + n_max
end

function inplace!(a, n_max, index_func)
    i1 = index_func(1, 2, 3; n_max=n_max)  # Using this version leads to allocations below
    # i1 = index_func(1, 2, 3)             # Using this version leads to 0 allocations
    i2 = size(a, 1) - 2i1
    for i in 1:i2                      # Allocates 3182688 B if using kwarg above
        a[i + i1] = a[i + i1 - 1]      # Allocates 9573120 B if using kwarg above
    end
    for i in 3:i2-4                    # Allocates 3182576 B if using kwarg above
        a[i + i1] -= a[i + i1 - 2]     # Allocates 12771408 B if using kwarg above
    end
end

function compute_a(n_max::Int64)
    a = randn(Float64, 100_000)
    inplace!(a, n_max, index)
    Profile.clear_malloc_data()
    inplace!(a, n_max, index)
end


compute_a(10)
```
And yes, there are plenty of ways to improve the performance of this simplified code with function barriers and such.  But my actual code is too complicated for that, with the kwarg func being used multiple times inside some loops.

If I look at the specializations of `inplace!(a, n_max, index)`, I get
```julia
svec(MethodInstance for inplace!(::Vector{Float64}, ::Int64, ::Function), MethodInstance for inplace!(::Vector{Float64}, ::Int64, ::typeof(index)), nothing, nothing, nothing, nothing, nothing, nothing)
```
That second element really looks to me like it specialized for my particular `index` function.

<details>
<summary>Here's all my versioninfo</summary>

```
julia> versioninfo()
Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.5.0)
  CPU: Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, haswell)
```

</details>

### LINKED ISSUE #45171: Incorrect line number for kwarg function call in stack trace
```
using DataFrames

a = [1, 2, 3]
b = [3, 4]
c() = DataFrame(:a => a, :b => b)
c()

#=
ERROR: DimensionMismatch: column :a has length 3 and column :b has length 2
Stacktrace:
 [1] DataFrame(columns::Vector{Any}, colindex::DataFrames.Index; copycols::Bool)
   @ DataFrames C:\Users\nicho\.julia\packages\DataFrames\zqFGs\src\dataframe\dataframe.jl:192
 [2] DataFrame(::Pair{Symbol, Vector{Int64}}, ::Vararg{Pair{Symbol, Vector{Int64}}}; makeunique::Bool, copycols::Bool)
   @ DataFrames C:\Users\nicho\.julia\packages\DataFrames\zqFGs\src\dataframe\dataframe.jl:247
 [3] DataFrame(::Pair{Symbol, Vector{Int64}}, ::Vararg{Pair{Symbol, Vector{Int64}}})
   @ DataFrames C:\Users\nicho\.julia\packages\DataFrames\zqFGs\src\dataframe\dataframe.jl:245
 [4] c()
   @ Main .\REPL[4]:1
 [5] top-level scope
   @ REPL[5]:1
=#
```
image

image

The screenshot is from C:\Users\nicho\.julia\packages\DataFrames\zqFGs\src\dataframe\dataframe.jl

Line 192 is where the error is thrown. Line 247 is the call site of the function that threw the error. Is Line 245 is supposed to be Line 243?

Another example, where the line number is only 1 off:
```
function aa()
       1
       2
       3
       error(3)
end

 function bb()
       3
       4
       aa()
end

function cc(;kwarg = true)
       1
       bb()
end

cc()

#=
ERROR: 3
Stacktrace:
 [1] error(s::Int64)
   @ Base .\error.jl:42
 [2] aa()
   @ Main .\REPL[1]:5
 [3] bb()
   @ Main .\REPL[2]:4
 [4] cc(; kwarg::Bool)
   @ Main .\REPL[3]:3
 [5] cc()
   @ Main .\REPL[3]:2
 [6] top-level scope
   @ REPL[4]:1
=#
```

### PR DESCRIPTION
**Pull request by KristofferC**
Backported PRs:
- [x] #44280 <!-- Distributed test suite: wrap another non-thread-safe test in `poll_while` -->
- [x] #44500 <!-- Improve inference for string operations -->
- [x] #44777 <!-- Pass around effects during inference cacheing -->
- [x] #44770 <!-- fix handling of `.op` in lowering -->
- [x] #44764 <!-- fix #44763, implement missing effects propagation from `abstract_invoke` -->
- [x] #44762 <!-- fix #44732, propagate effects of a thrown concrete call correctly -->
- [x] #44749 <!-- fix `===` when encountering null pointer -->
- [x] #44651 <!-- Fix performance issue with diagonal multiplication -->
- [x] #44781 <!-- profile: fix async deadlock -->
- [x] #44842 <!-- Partially revert PR 42703 to fix embedding with MSVC -->
- [x] #44810 <!-- [Zlib_jll] Update to v1.2.12+3 -->
- [x] #44797 <!-- fix missing field type initialization vars -->
- [x] #44807 <!-- Fix "anonymous types declared in an anonymous union" warnings -->
- [x] #44942 <!-- Properly set Vboxed field during union codegen -->
- [x] #44940 <!-- Update rotted Distributed, noteworthy external packages docs -->
- [x] #44956 <!-- asyncevents: fix missing GC root and race -->
- [x] #44972 <!-- [REPL] Fix a REPL test failure by removing an erroneous space in test (introduced by #33805) -->
- [x] #44976 <!-- Fix embedding with MSVC -->
- [x] #45026 <!-- Slight update to doc for PDF to render correctly -
- [x] #44786 <!-- effects: change the `overlayed::Bool` property to `nooverlayed::Bool` -->
- [x] #44785 
- [x] #44808
- [x] #44775 
- [x] #44761 <!-- fix #44735, assert `typeintersect` arguments on Julia level -->
- [x] #44966 
- [x] #45017 
- [x] #44804 <!-- follow up #44786, fix `findsup` to return correct `overlayed` value -->
- [x] #44699 <!-- Initialize jl_page_size when not calling julia_init. -->
- [x] #44793 <!-- Change type of donotdelete intrinsics -->
- [x] #45030 <!-- inference: relax backedge optimization condition -->
- [x] #45035 <!-- gc-ext: only sweep unmarked objects -->
- [x] #45034 <!-- [doc] Add a Cover page to PDF doc -->
- [x] #45071 <!-- Correct man-custom-indices ref in devdocs -->
- [x] #45072 <!-- Correct typo in devdocs gc-sa -->
- [x] #45119 <!-- [deps] Fix compillation of LLVM with system zlib -->
- [x] #45048 <!-- Create a copy while evaluating eigvals(::Diagonal) -->
- [x] #45009 <!-- Fix bug in `pinv` -->
- [x] #45176 <!-- [deps] Remove Python workaround in LLVM build, no longer necessary -->
- [x] #45149 <!-- Fix different .cachepath initialization points for precompile load -->
- [x] #45214 <!-- Avoid race condition in cleaning up cache files -->
- [x] #44767 <!-- Consider additional default ssh keys for LibGit2. -->
- [x] #45199 <!-- fix #45171, keyword sorter location should just be definition line -->
- [x] #45198 <!-- fix #45162, function arg not specialized if only called with keywords -->
- [x] #44534 <!-- Fix a concurrency bug in `iterate(::Dict)` -->
- [x] #44840 <!-- Bump Documenter to get the new pdf improvements for the manual. -->
- [x] #45155 <!-- codegen: add handling for undefined phinode values -->
- [x] #45289 <!-- Fix `copyto!(dest, Rdest::CartesianIndices, src, Rsrc::CartesianIndices)` with different step.  -->
- [x] #45333 <!-- Fix typo in #45289. -->
- [x] #45189 <!-- Resurrect libunwind patches -->
- [x] #45346 <!-- [deps] Use newer `config.sub` when building nghttp2 -->
- [x] #45344 <!-- fix #45024, lost `expected assignment after const` error -->
- [x] #45249 <!-- codegen: explicitly handle Float16 intrinsics -->
- [x] #45365 <!-- Fix error in validating complex row-first hvncat -->
- [x] #45375 <!-- Apply patch for GMP CVE-2021-43618 -->
- [x] #45392 <!-- Use root module when determining UUID in @artifact_str -->
- [x] #45391 <!-- OpenBLAS: Find objconv in its proper path -->
- [x] #45398 <!-- Update PCRE2 to 10.40 -->
- [x] #45412 <!-- set default blas num threads to Sys.CPU_THREADS / 2 in absence of OPENBLAS_NUM_THREADS -->
- [x] #45420 <!-- Don't error when transposing a single character -->
- [x] #45435 <!-- Update p7zip to 17.04 -->
- [x] #44881 <!-- [Distributed] Set stdin to devnull before closing it -->
- [x] #45016 <!-- Fix use-after-free bugs in debuginfo -->
 
Contains multiple commits, manual intervention needed:
- [ ] #44360 <!-- Simplification of libblastrampoline stuff in LinearAlgebra -->

Non-merged PRs with backport label:
- [ ] #45370 <!-- Disallow reinterpreting a non-singleton array into a singleton type -->
- [ ] #45120 <!-- Remove duplicate definition of jl_gc_safepoint -->
- [ ] #45001 <!-- Avoid unnecessary invalidations for `invoke`d fallbacks -->
- [ ] #44507 <!-- Extend `strides` for `ReshapedArray` with strided parent. -->

### CODE REVIEW

**Comment by KristofferC** [comment]
@nanosoldier `runtests(["AbstractDifferentiation", "AbstractTrees", "AdaStress", "AdventOfCode", "AnyMOD", "ApproxFun", "ApproxFunFourier", "ApproxFunOrthogonalPolynomials", "ApproxFunSingularities", "AssociativeArrays", "AtomicGraphNets", "AxisKeys", "AxisSets", "AxisTables", "BEASTXMLConstructor", "BIDSTools", "BPFnative", "BSON", "CBinding", "CGAL", "CUDAKernels", "Caching", "Causal", "ChainRulesCore", "ChainRulesOverloadGeneration", "ClimaCorePlots", "ClimaCoreVTK", "ClimatePlots", "ClimateTools", "Cloudy", "CombinedParsers", "CompactBases", "CompilerPluginTools", "Conductor", "CxxWrap", "DailyTreasuryYieldCurve", "Dash", "DataFrames", "DataStructures", "DeIdentification", "DeepDiffs", "DiffEqCallbacks", "DiscreteEvents", "Dispatcher", "DynamicBoundsBase", "DynamicalBilliards", "EconJobMarket", "Enzyme", "EquationsOfStateOfSolids", "ExSup", "FastJet", "FeatureEng", "FinEtoolsVoxelMesher", "FindClosest", "FlightMechanics", "FlxQTL", "FormattedTables", "GeneralizedSVD", "GenericSchur", "GeoEfficiency", "GeoStatsDevTools", "GoogleCodeSearch", "GroebnerBasis", "HarmonicOrthogonalPolynomials", "HierarchicalUtils", "HomotopyContinuation", "Hypatia", "IBMQClient", "ImageCore", "ImageQuilting", "ImageSegmentation", "ImplicitArrays", "InverseLaplace", "JET", "JSONPointer", "JuliennedArrays", "KernelGradients", "LCIO", "LMDB", "LRSLib", "LazyArrays", "LegibleLambdas", "LiterateOrg", "LogParser", "LogRoller", "LoweredCodeUtils", "MEstimation", "MIRT", "MIRTjim", "MambaModels", "MappedArrays", "MarketCycles", "Memento", "MeshCore", "MetaArrays", "Metatheory", "MinimallyDisruptiveCurves", "MultinomialRegression", "MultipleTesting", "NMRTools", "NeuralArithmetic", "NicePipes", "NotebookToLaTeX", "OpenEphysLoader", "OscillatoryIntegrals", "PLCTag", "POMDPPolicies", "PProf", "PSIS", "PackageAnalyzer", "ParallelUtilities", "Parallelism", "Pathfinder", "PkgDeps", "Plots", "Poltergeist", "Polyhedra", "PredictMDExtra", "ProfileSVG", "QuantizedArrays", "QuantumOptics", "ROCKernels", "RSCG", "RedefStructs", "RiemannHilbert", "RobotDescriptions", "RobotOSData", "Runner", "SIAN", "SOM", "SemiseparableMatrices", "Signals", "SimplePadics", "SingularIntegralEquations", "SnoopCompile", "SocialSolver", "Soss", "SparseTimeSeries", "SpatialBoundaries", "StableDQMC", "StackOverflow", "StarAlgebras", "StatProfilerHTML", "StaticTrafficAssignment", "StochasticDiffEq", "StrRegex", "StructuralIdentifiability", "StructuredArrays", "TORA", "TensorKitManifolds", "Tracker", "TraitSimulation", "TurbulenceConvection", "ZfpCompression", "Zomato"], vs = ":release-1.7")`

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1102745220) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/5073c4e_vs_857b62d/report.html).

**Comment by KristofferC** [comment]
@nanosoldier `runtests(ALL, vs = ":release-1.7")`

**Comment by KristofferC** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":release-1.7")`

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1131259552):
```
NanosoldierError: error when preparing/pushing to report repo: failed process: Process(setenv(`git push`; dir="/nanosoldier/workdir/NanosoldierReports"), ProcessExited(1)) [1]

```
Unfortunately, the logs could not be uploaded.

**Comment by oscardssmith** [comment]
https://github.com/JuliaLang/julia/pull/41200 isn't going to be back-ported (if we backported performance increases, we'd never release). The other does need backporting.

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1131259470) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/eb82f18_vs_742b9ab/report.html).

**Comment by KristofferC** [comment]
@nanosoldier `runtests(["AbstractDifferentiation", "AbstractGPs", "AbstractPlotting", "AbstractTrees", "AcousticRayTracers", "AdaStress", "AdventOfCode", "AlphaStableDistributions", "Altro", "AnyMOD", "ArgoData", "AssociativeArrays", "AxisKeys", "AxisSets", "AxisTables", "BIDSTools", "BSON", "Bijectors", "BlsData", "Books", "CDDLib", "CGAL", "CMAEvolutionStrategy", "CUDAKernels", "CachePath", "Caching", "Caesar", "Causal", "ChainRules", "ChainRulesCore", "ChainRulesOverloadGeneration", "ClimaCoreTempestRemap", "ClimatePlots", "ClimateTools", "CombinedParsers", "CompactBases", "CompilerPluginTools", "Conductor", "CxxWrap", "DIVAnd", "DataAssim", "DataFrames", "DeIdentification", "DebugAdapter", "DecisionProgramming", "DeepCompartmentModels", "DeepDiffs", "DelayDiffEq", "DescriptorSystems", "DiffEqCallbacks", "DiscreteEvents", "Dispatcher", "DistributedArrays", "DomainSets", "DynamicBoundsBase", "DynamicPPL", "DynamicalBilliards", "EAGO", "EconJobMarket", "Enzyme", "EquationsOfStateOfSolids", "Evolutionary", "ExSup", "Exodus", "Express", "FHIRClient", "FSInteraction", "FactorGraph", "FastJet", "FeatureEng", "FinEtoolsVoxelMesher", "FindClosest", "FindSteadyStates", "Flamenco", "FlightMechanics", "FlxQTL", "FormattedTables", "ForwardDiffPullbacks", "FrankWolfe", "GEOTRACES", "GadgetIO", "GeneralizedSVD", "GeoDatasets", "GeoEfficiency", "GeoStatsDevTools", "GoogleCodeSearch", "GroebnerBasis", "HomotopyContinuation", "Hypatia", "IBMQClient", "ITensors", "ImageCore", "ImageQuilting", "ImageSegmentation", "ImplicitArrays", "IntervalTrees", "Intervals", "InvariantCausal", "InverseLaplace", "JET", "JSONPointer", "JuliennedArrays", "KernelFunctions", "KernelGradients", "KiteModels", "Knet", "LCIO", "LMDB", "LRSLib", "LabelledArrays", "LegibleLambdas", "LiterateOrg", "LogParser", "LogRoller", "LoweredCodeUtils", "MCPhylo", "MEstimation", "MIRTjim", "MITgcmTools", "MPSKit", "MambaModels", "MappedArrays", "MarketCycles", "MatrixPencils", "Memento", "MemoizedMethods", "MeshCore", "MetaArrays", "Metatheory", "MiniLogging", "Modia3D", "MultinomialRegression", "MultipleTesting", "MultiplesOfPi", "NCDatasets", "NMRTools", "NativeNaNMath", "NetCDF", "NetcdfIO", "NicePipes", "NotebookToLaTeX", "NumericalAlgorithms", "Oceanostics", "OpenEphysLoader", "OptimalTransmissionRouting", "OteraEngine", "PALEOboxes", "PALEOmodel", "POMDPPolicies", "PSIS", "PackageAnalyzer", "ParallelUtilities", "Parallelism", "ParameterEstimocean", "ParameterSpacePartitions", "PartialSvdStoch", "PerfChecker", "Photodynamics", "PhysOcean", "PkgDeps", "PlanningDomains", "PlotPlants", "Poltergeist", "Polyhedra", "PowerGraphics", "PredictMDExtra", "ProbNumDiffEq", "ProfileSVG", "QHull", "Qaintessent", "QuantizedArrays", "ROCKernels", "RRTMGP", "RSCG", "RadiativeTransfer", "RankRevealing", "RedefStructs", "ResizingTools", "RiemannHilbert", "Rimu", "RobotDescriptions", "RobotOSData", "Runner", "SIAN", "SOM", "SampledSignals", "Scheduling", "ShallowWaters", "Signals", "SimplePadics", "SlackThreads", "SnoopCompile", "SocialSolver", "Soss", "SparseTimeSeries", "SpatialBoundaries", "SpatialJackknife", "SpeedyWeather", "StableDQMC", "StackOverflow", "StarAlgebras", "StatProfilerHTML", "StaticTools", "StaticTrafficAssignment", "StorageMirrorServer", "StrRegex", "StructuralIdentifiability", "StructuredArrays", "SuiteSparseMatrixCollection", "Sundials", "SurfaceFluxes", "SwitchOnSafety", "TORA", "TensorGames", "TestReports", "Thermodynamics", "Tracker", "TurbulenceConvection", "VMEC", "Vlasiator", "Wflow", "WorldOceanAtlasTools", "Xyce", "YAXArrayBase", "ZfpCompression", "Zomato"], vs = ":release-1.7")`

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1132199873) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/eb82f18_vs_742b9ab/report.html).

**Comment by ufechner7** [comment]
What is blocking this PR?

**Comment by KristofferC** [comment]
> What is blocking this PR?

There were some dependency updates that fixes security issues that should go in before RC. But should be in pretty good shape now.

**Comment by KristofferC** [comment]
To check for regressions in this PR itself:

@nanosoldier `runtests(ALL, vs = "release-1.8")`

**Comment by KristofferC** [comment]
To check for regressions in this PR itself:

@nanosoldier `runtests(ALL, vs = ":release-1.8")`

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1135053157) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/6acc46b_vs_3e092a2/report.html).

**Comment by KristofferC** [comment]
@nanosoldier `runtests(["AbstractAlgebra", "AdvancedVI", "BayesianQuadrature", "CachePath", "ChainRules", "CropRootBox", "FHIRClient", "GaussianMixtureAlignment", "HighDimPDE", "HydrophoneCalibrations", "KissMCMC", "Knet", "MatrixEquations", "MatrixProfile", "MeshMaker", "PowerSimulationsDynamics", "PyCallJLD", "QuantumTomography", "Relief", "ResizingTools", "SlackThreads", "StoppingInterface", "TensorKit", "TuringGLM"], vs = ":release-1.8")`

File: stdlib/Pkg.version [hunk 1]
```diff
@@ -1,4 +1,4 @@
 PKG_BRANCH = master
```
Comment (IanButterworth) [inline stdlib/Pkg.version:1]: I think the branch should be release-1.8?

I'm not sure it matters though?

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1135454178) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/6acc46b_vs_3e092a2/report.html).

File: stdlib/Pkg.version (hunk 1 above)
Comment (KristofferC) [reply stdlib/Pkg.version:1]: I think it matters when you do `DEPS_GIT`. I'll update it.

**Comment by maleadt** [comment]
Pushed a manual backport of https://github.com/JuliaLang/julia/pull/45016. Noticed while doing so that https://github.com/JuliaLang/julia/pull/44967 wasn't marked for backport, maybe we need to include it.

**Comment by vtjnash** [comment]
https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/2eaa6b9_vs_742b9ab/report.md

**Comment by KristofferC** [comment]
@nanosoldier `runtests(["AbstractAlgebra", "AdvancedVI", "BayesianQuadrature", "CachePath", "ChainRules", "CropRootBox", "FHIRClient", "GaussianMixtureAlignment", "HighDimPDE", "HydrophoneCalibrations", "KissMCMC", "Knet", "MatrixEquations", "MatrixProfile", "MeshMaker", "PowerSimulationsDynamics", "PyCallJLD", "QuantumTomography", "Relief", "ResizingTools", "SlackThreads", "StoppingInterface", "TensorKit", "TuringGLM"], vs = ":release-1.8")`

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/44789#issuecomment-1138209897) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/8bca2f4_vs_3e092a2/report.html).

**Pull request merged**